dev-lang/nim/nim-0.17.0.recipe

```diff
@@ -0,0 +1,103 @@
+SUMMARY="An imperative, multi-paradigm, compiled programming language"
+DESCRIPTION="Nim is a systems and applications programming language. \
+Statically typed and compiled, it provides unparalleled performance in an \
+elegant package.
+
+Features:
+* High-performance garbage-collected language
+* Compiles to C, C++ or JavaScript
+* Produces dependency-free binaries
+* Runs on Windows, macOS, Linux, and more"
+HOMEPAGE="https://nim-lang.org/"
+COPYRIGHT="2006-2016 Andreas Rumpf"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://nim-lang.org/download/nim-$portVersion.tar.xz" # GitHub release unsuited for us
+CHECKSUM_SHA256="36e18dd9384f6c67e6d0199b871b43e774a0af30532698184d6f5a9cc9ac7a9b"
+SOURCE_URI_2="https://github.com/nim-lang/csources/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256_2="6b8de28dc317534e45366b62dcad14b9932c0de398c44e1506f5bbd80019c671"
+SOURCE_FILENAME_2="csources-$portVersion.tar.gz"
+ADDITIONAL_FILES="nim.rdef.in"
+PATCHES="nim-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	nim$secondaryArchSuffix = $portVersion
+	cmd:nim$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libpcre$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcrypto$secondaryArchSuffix
+	devel:libpcre$secondaryArchSuffix
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/nim/nim.cfg keep-old
+	settings/nim/nimdoc.cfg keep-old
+	settings/nim/nimdoc.tex.cfg keep-old
+	"
+
+PATCH()
+{
+	sed -i "s|@BINDIR@|$binDir|" install.sh
+	sed -i "s|@CONFIGDIR@|$settingsDir/nim|" install.sh
+	sed -i "s|@LIBDIR@|$libDir/nim|" install.sh
+	sed -i "s|@DOCDIR@|$docDir|" install.sh
+	sed -i "s|@DATADIR@|$dataDir/nim|" install.sh
+	sed -i "s|@NIMBLEDIR@|$dataDir/nim|" install.sh
```

Comment:
Warning for such "updates": the revision number is included in $libDir or $binDir, thus when bumping the revision number, the sources are already patches, install.sh won't be updated with the new revision.